src/bin/cargo/cli.rs

```diff
@@ -699,7 +699,7 @@ See '<bright-cyan,bold>cargo help</> <cyan><<command>></>' for more information
             .action(ArgAction::SetTrue)
             .global(true)
             .hide(true))
-        .arg(multi_opt("config", "KEY=VALUE|PATH", "Override a configuration value").global(true))
+        .arg(multi_opt("config", "KEY=VALUE|PATH", "Override a configuration value").value_hint(clap::ValueHint::FilePath).global(true))
```

Comment:
Maybe one thing worth discussion. According to [the doc](https://doc.rust-lang.org/nightly/cargo/reference/config.html#config-relative-paths):

> Note: To maintain consistency with existing .cargo/config.toml probing behavior, it is by design that a path in a config file passed via --config <path> is also relative to two levels up from the config file itself.

So personally would expect people usually put their extra config under `.cargo/` or `.config/`. Not sure if this means that we might want to have custom completer

Reply:
If we `filter` for these, they'll be shown first but all other dirs will still be shown but after. 

Whats more of a problem is that it hides the candidates of hidden files and directories. In that state, they are only shown if there are no visible candidates.

I might want to add an override for this in clap. It would be easy to say "don't hide for this completer" but that will then show everything when likely only a subset is wanted. I could have a separate `hide` function that takes a predicate to override the default hide behavior.